system-contracts/contracts/EvmInterpreter.sol

```diff
@@ -2335,16 +2342,15 @@ contract EvmInterpreter {
         address addr = address(EVM_GAS_MANAGER);
         assembly {
             mstore(0, selector)
-
-            let success := call(gas(), addr, 0, 0, 4, 0, 64)
-
+        }
+        bool success = SystemContractsCaller.rawSystemCall(uint32(gasleft()), addr, 0, 0, 4, true);
+        assembly {
+            _passGas := mload(0)
+            isStatic := mload(32)
```

Comment:
formally the behavior is the same between the following two codes:

```
            _passGas := mload(0)
            isStatic := mload(32)
            if iszero(success) {
                // This error should never happen
                revert(0, 0)
            }
```
and
```
            if iszero(success) {
                // This error should never happen
                revert(0, 0)
            }
            _passGas := mload(0)
            isStatic := mload(32)
```

but the latter is better because we fail as fast as possible and do not continue execution with invalid data

Reply:
fixed